arrow-integrations-retrofit-adapter/src/main/kotlin/arrow/integrations/retrofit/adapter/CallKindAdapterFactory.kt

```diff
@@ -1,36 +1,4 @@
 package arrow.integrations.retrofit.adapter
 
-import retrofit2.CallAdapter
-import retrofit2.Retrofit
-import java.lang.reflect.ParameterizedType
-import java.lang.reflect.Type
-
-class CallKindAdapterFactory : CallAdapter.Factory() {
-
-  companion object {
-    fun create(): CallKindAdapterFactory = CallKindAdapterFactory()
-  }
-
-  override fun get(returnType: Type, annotations: Array<Annotation>, retrofit: Retrofit): CallAdapter<*, *>? {
-    val rawType = CallAdapter.Factory.getRawType(returnType)
-
-    if (returnType !is ParameterizedType) {
-      val name = parseTypeName(returnType)
-      throw IllegalArgumentException("Return type must be parameterized as " +
-        "$name<Foo> or $name<out Foo>")
-    }
-
-    val effectType = CallAdapter.Factory.getParameterUpperBound(0, returnType)
-
-    return if (rawType == CallK::class.java) {
-      CallKind2CallAdapter<Type>(effectType)
-    } else {
-      null
-    }
-  }
-}
-
-private fun parseTypeName(type: Type) =
-  type.toString()
-    .split(".")
-    .last()
+@Deprecated("Please use ArrowCallAdapterFactory instead", ReplaceWith("ArrowCallAdapterFactory"))
+typealias CallKindAdapterFactory = ArrowCallAdapterFactory
```

Comment:
IMO doesn't make sense to have multiple factories, so I named the one true factory `ArrowCallAdapterFactory`. Old name is now a typealis and deprecated.

Reply:
Reverted this change.